.travis.yml

```diff
@@ -90,7 +88,15 @@ script:
   - if [[ $DEVITO_BACKEND != 'yask' ]]; then py.test -vs --nbval examples/seismic/tutorials; fi
   - if [[ $DEVITO_BACKEND != 'yask' ]]; then py.test -vs --nbval examples/cfd; fi
 
-  # Docs generation and deployment
+  # Docs generation
   - sphinx-apidoc -f -o docs/ examples
   - sphinx-apidoc -f -o docs/ devito devito/yask/*
-  - if [[ $DEVITO_ARCH == 'gcc-5' ]]; then ./docs/deploy.sh; fi
+  # Docs deployment only once from a single build
+  - if [[ $DEVITO_ARCH == 'gcc-5' ]]; then
+      set -e;
+      pip install doctr;
+      cd docs;
+      make html;
+      cd ..;
+      doctr deploy . --built-docs docs --key-path .deploy_key.enc;
```

Comment:
Does `doctr` know not to deploy from branches/PRs?

Reply:
it automatically deploys to ``gh-pages`` if that's what you're asking?

Reply:
I'm asking if every travis build run, including branches and PRs, is executing a push to the branch (dangerous and racy) or if `doctr` knows only to deploy if the branch we're on is `master`. I'm assuming it does, I'd just like to know that someone double-checked this.

Reply:
Oh, I see. so I had a look at ``doctr deploy --help``. Amongst the things it says:
```
...
  --branch-whitelist [BRANCH [BRANCH ...]]
                        Branches to deploy from. Pass no arguments to not
                        build on any branch (typically used in conjunction
                        with --build-tags). Note that you can deploy from
                        every branch with --no-require-master.
  --no-require-master   Allow docs to be pushed from a branch other than
                        master
  --require-master      Inverse of "--no-require-master"
...
```
I'm not sure how to interpret this, as I cannot see what's the default option. Should I add ``--require-master`` ?

I couldn't find a definitive answer here either: https://github.com/drdoctr/doctr

I also went looking for what SymPy does: https://github.com/sympy/sympy/blob/master/.travis.yml#L285
they deploy conditionally on whether they're testing sphinx, but I cannot see any master-specific instruction

Reply:
Sounds to me like it does the right thing by default, so I'm happy to try this as is. Good to go from me.